src/app/core/services/observer-trips/observer-trips.service.ts

```diff
@@ -0,0 +1,127 @@
+import { HttpErrorResponse, HttpStatusCode } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { FeatureCollection } from '@turf/turf';
+import { BehaviorSubject, combineLatest, distinctUntilChanged, firstValueFrom, from, map, mapTo, Observable, ReplaySubject, skipWhile, switchMap } from 'rxjs';
+import { RegobsAuthService } from 'src/app/modules/auth/services/regobs-auth.service';
+import { TripService } from 'src/app/modules/common-regobs-api';
+import { LoggingService } from 'src/app/modules/shared/services/logging/logging.service';
+import { DatabaseService } from '../database/database.service';
+
+const msOneWeek = 604800000;
+const dataKey = 'REGOBS_OBSTRIPS_GEOJSON';
+const toggleKey = 'REGOBS_OBSTRIPS_ON';
+const dataModifiedKey = 'REGOBS_OBSTRIPS_CHANGED_DATE';
+
+const shouldUpdate = (mTime: number) => {
+  const msSinceUpdate = Date.now() - mTime;
+  return msSinceUpdate > msOneWeek;
+};
+
+const DEBUG_TAG = 'ObserverTrips';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ObserverTripsService {
+
+  geojson: Observable<FeatureCollection | null>;
+  canShow = new ReplaySubject(1);
+  toggleOn = new BehaviorSubject(false);
+
+  constructor(
+    private tripService: TripService,
+    private logger: LoggingService,
+    private dbService: DatabaseService,
+    authService: RegobsAuthService
+  ) {
+    this.geojson = combineLatest([
+      authService.loggedInUser$,
+      this.toggleOn.pipe(distinctUntilChanged())
+    ]).pipe(
+      skipWhile(([o,]) => !o.isLoggedIn),
+      switchMap(([o,]) => o.isLoggedIn ? from(this.getData()) : from(this.clear()).pipe(mapTo(null))),
```

Comment:
Jeg synes ikke dette er super-intuitivt, men sånn er det bare. Skjønner at det er kjekt for de som bruker kartlaget at det er en Observable